### PR TITLE
Fixes #465 Maps without author can be updated by anyone

### LINKED ISSUE #465: Maps without author can be updated by anyone

### PR DESCRIPTION
**Pull request by bukajsytlos**
Fixes #465 Maps without author can be updated by anyone

### CODE REVIEW

File: src/main/java/com/faforever/api/map/MapService.java [hunk 1]
```diff
@@ -343,11 +342,9 @@ private void validateScenarioLua(MapLuaAccessor mapLua, MapNameBuilder mapNameBu
     Optional<Map> existingMapOptional = mapRepository.findOneByDisplayName(displayName);
     existingMapOptional
       .ifPresent(existingMap -> {
-        Optional.ofNullable(existingMap.getAuthor())
-          .filter(existingMapAuthor -> !Objects.equals(existingMapAuthor, author))
-          .ifPresent(existingMapAuthor -> {
-            throw ApiException.of(ErrorCode.MAP_NOT_ORIGINAL_AUTHOR, existingMap.getDisplayName());
-          });
+        final Player existingMapAuthor = existingMap.getAuthor();
+        if (existingMapAuthor == null || !existingMapAuthor.equals(author))
```
Comment (Sheikah45) [inline src/main/java/com/faforever/api/map/MapService.java]: Objects.equals(existingMapAuthor, author)

Comment (bukajsytlos) [reply src/main/java/com/faforever/api/map/MapService.java]: I don't want nulls to be equal just for sure

Comment (Brutus5000) [reply src/main/java/com/faforever/api/map/MapService.java]: It's also more expressively, as it indicates that existingMapAuthor can be null

Comment (Brutus5000) [inline src/main/java/com/faforever/api/map/MapService.java]: I think it was code style convention to always use brackets for if/else even for one liners

**Review by Brutus5000 — APPROVED** [review]

File: src/main/java/com/faforever/api/map/MapService.java (hunk 1 above)
Comment (bukajsytlos) [reply src/main/java/com/faforever/api/map/MapService.java]: yeah not sure how this has happened :), I don't like bracketless ifs too

Comment (Sheikah45) [reply src/main/java/com/faforever/api/map/MapService.java]: Yeah all good I had thought there was an explicit null check on author that threw an error if null coming into the function but it isn't a big deal

**Pull request merged**